src/language-js/printer-estree.js

```diff
@@ -4107,11 +4107,11 @@ function printMemberChain(path, options, print) {
   //     .map(x => x)
   //
   // In order to detect those cases, we use an heuristic: if the first
-  // node is just an identifier with the name starting with a capital
-  // letter, just a sequence of _$ or this. The rationale is that they are
-  // likely to be factories.
+  // node is an identifier with the name starting with a capital letter,
+  // containing only _ and $, or shorter than tabWidth. The rationale
+  // is that they are likely to be factories.
   function isFactory(name) {
-    return name.match(/(^[A-Z])|^[_$]+$/);
+    return name.match(/(^[A-Z])|^[_$]+$/) || name.length <= options.tabWidth;
```

Comment:
You can get rid of the special case for _$ since it's always going to be <= tabWidth

Reply:
@vjeux Note the `+` — `$$$$$`, `_____`, and `$_$_$` would all count as factory functions, but still be longer than `tabWidth`.

Reply:
If you write this kind of variables, I'm not going to feel sorry for you :p

Reply:
I agree it’s not a good idea, but I wanted to point it out anyway in case you missed an edge case.

Reply:
Removed the `$` special case; `$_$` no longer counts as a factory function.